### PR TITLE
[NUI] Add WidgetComponent for ComponentBasedApplication

### PR DESCRIPTION
**Pull request by tscholb**
Adds implementations and samples for WidgetComponent use

For more information, please refer to the links below
- Tizen.Applications.ComponentBased: https://github.com/Samsung/TizenFX/tree/master/src/Tizen.Applications.ComponentBased
- Tizen.Applications.ComponentBased.Default: https://github.com/Samsung/TizenFX/tree/master/src/Tizen.Applications.ComponentBased.Default/Tizen.Applications.ComponentBased.Default
- EFL Sample app: https://github.sec.samsung.net/appfw/csharp-component-based-app

### API Changes ###

 - ACR:

### CODE REVIEW

**Review by hinohie — APPROVED** [review]
Looks good to me

**Review by taehyub — APPROVED** [review]
LGTM 💯

File: src/Tizen.NUI/src/internal/Common/NUIWindowProxy.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright(c) 2021 Samsung Electronics Co., Ltd.
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/internal/Common/NUIWindowProxy.cs:2]: Very minor 
```suggestion
 * Copyright(c) 2024 Samsung Electronics Co., Ltd.
```

File: src/Tizen.NUI/src/public/Application/NUIWidgetComponent.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        /// </summary>
+        /// <returns>True if a service component is successfully created</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool OnCreate(int width, int height)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Overrides this method if want to handle behavior when the component receives the appcontrol message.
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Application/NUIWidgetComponent.cs:79]:
```suggestion
        /// Overrides this method if you want to handle behavior when the component receives the appcontrol message.
```

File: test/Tizen.NUI.WidgetViewTest/4.WidgetComponentTest/Tizen.NUI.WidgetViewTest/SimpleWidgetViewApp.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd.
```
Comment (Seoyeon2Kim) [inline test/Tizen.NUI.WidgetViewTest/4.WidgetComponentTest/Tizen.NUI.WidgetViewTest/SimpleWidgetViewApp.cs:2]:
```suggestion
 * Copyright (c) 2024 Samsung Electronics Co., Ltd.
```

**Review by Seoyeon2Kim — COMMENTED** [review]
Overall Looks good because WidgeView is already specific for the owner ;)
And adding samples is also great/

File: test/Tizen.NUI.WidgetViewTest/4.WidgetComponentTest/Tizen.NUI.WidgetViewTest/SimpleWidgetViewApp.cs (hunk 3 above)
Comment (tscholb) [reply test/Tizen.NUI.WidgetViewTest/4.WidgetComponentTest/Tizen.NUI.WidgetViewTest/SimpleWidgetViewApp.cs:2]: Thank you for your review. I'll fix it quickly 😊

**Review by Seoyeon2Kim — APPROVED** [review]
LGTM!!!!!!!!

**Comment by tscholb** [comment]
리뷰 남겨주신 모든분들 감사합니다! 반영하겠습니다 👍

**Pull request merged**